design-proposals/vm-memory-snapshot.md

```diff
@@ -0,0 +1,151 @@
+# Overview
+
+Proposal for taking VM RAM snapshot which can be either a stand alone action, or alongside Online VMSnapshot.
+
+## Motivation
+One of the methods of troubleshooting guest virtual machines is getting a memory dump and analyzing it. As this is a standard capability in most VM management systems, it is also required for kubevirt’s VMs.
+
+## Goals
+Have a mechanism to take and get a VM memory snapshot.
+This mechanism can be either during the Online VMSnapshot process to get a complete snapshot of a VM: disks and Ram alike, or as a stand alone command to get the memory dump only.
+
+## Non Goals
+Currently the memory snapshot will not be used for VM hibernation, not in the VM restore process.
+
+## User Stories
+As a Kubeviert user I would like to take a snapshot of a running VM so that I can later inspect its disk and memory
```

Comment:
```suggestion
As a KubeVirt user I would like to take a snapshot of a running VM so that I can later inspect its disk and memory
```
spelling error